api/fhda_api_1.1.yaml

```diff
@@ -0,0 +1,426 @@
+
+  license:
+    name: License Not determined
+    url: http://www.blank.org
+
+paths:
+#################COURSE OPERATIONS###########################
+  /course/get:
+    get:
+      summary: get course info by passing parameters
+      operationId: getCourse
+      description: |
+        Search for course by quarter
+      produces:
+      - application/json
+      parameters:
+        - in: query
+          name: quarter
+          description: the quarter for searching
+          type: string
+          required: true
+        - in: query
+          name: crn
+          description: the crn for searching
+          type: string
+          required: false
+      responses:
+        200:
+          description: search results matching criteria
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/CourseAcctD001A'
+        400:
+          description: bad input parameter
+        404:
+          description: cannot find the course with given parameter
+          
+  /course/add:
+    post:
+      summary: adds a course
+      operationId: addCourse
+      description: Adds a course to the system
+      parameters:
+      - in: body
+        name: course
+        description: course object to add
+        schema:
+          $ref: '#/definitions/CourseAcctD001A'
+      responses:
+        201:
+          description: course created
+        400:
+          description: invalid input, object invalid
+        409:
+          description: an existing course already exists
+          
+#################DEPTARTMENT OPERATIONS###########################
+  /dept/get:
+    get:
+      summary: get deptartment info
+      operationId: getDept
+      description: |
+        Search for department given input department name
+      produces:
+      - application/json
+      parameters:
+      - in: query
+        name: deptName
+        description: the department name for searching
+        type: string
+        required: true
+      responses:
+        200:
+          description: search results matching criteria
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/DepartmentAcct'
+        400:
+          description: bad input parameter
+        404:
+          description: cannot find the department with given parameter
+          
+  /dept/add:
+    post:
+      summary: adds a department
+      operationId: addDept
+      description: Adds a department to the system
+      parameters:
+      - in: body
+        name: department
+        description: departmentr object to add
+        schema:
+          $ref: '#/definitions/DepartmentAcct'
+      responses:
+        201:
+          description: department created
+        400:
+          description: invalid input, object invalid
+        409:
+          description: an existing department already exists
+          
+
+#################Instructor OPERATIONS###########################
+  /instructor/get:
```

Comment:
Would it be better to have a URL as “/instructor”, the method as "GET"(for retrieving), "POST"(for adding), etc?

Reply:
https://www.restapitutorial.com/lessons/httpmethods.html